### PR TITLE
Add RGB support to VulkanDriver.

### PR DESCRIPTION
**Pull request by prideout**

### CODE REVIEW

File: filament/src/driver/DataReshaper.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_FILAMENT_DRIVER_DATARESHAPER_H
+#define TNT_FILAMENT_DRIVER_DATARESHAPER_H
+
+#include <stddef.h>
+
+namespace filament {
```
Comment (bejado) [inline filament/src/driver/DataReshaper.h]: Should we nest this under the `driver` namespace?

File: filament/src/driver/DataReshaper.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    static void reshape(void* dest, const void* src, size_t numSrcBytes) {
+        const componentType* in = (const componentType*) src;
+        componentType* out = (componentType*) dest;
+        const size_t numWords = numSrcBytes / numSrcChannels;
+        for (size_t word = 0; word < numWords; ++word) {
+            for (size_t component = 0; component < numSrcChannels; ++component) {
+                out[component] = in[component];
+            }
+            for (size_t component = numSrcChannels; component < numDstChannels; ++component) {
+                out[component] = 0;
```
[the remark this answers is not in the document]
Comment (prideout) [reply filament/src/driver/DataReshaper.h]: Done, good catch.

File: filament/src/driver/DataReshaper.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+namespace filament {
+
+// This little utility adds padding to multi-channel interleaved data by inserting dummy values.
+// This is very useful for platforms that only accept 4-component data, since users often wish to
+// submit 3-component data.
+template<typename componentType, int numSrcChannels, int numDstChannels>
+class DataReshaper {
+public:
+    // The signature of reshape mimics the signature of memcpy.
+    static void reshape(void* dest, const void* src, size_t numSrcBytes) {
```
[the remark this answers is not in the document]
Comment (prideout) [reply filament/src/driver/DataReshaper.h]: Done.

File: filament/src/driver/DataReshaper.h (hunk 1 above)
Comment (prideout) [reply filament/src/driver/DataReshaper.h]: Well, the other shared classes in this folder don't seem to live in the driver namespace, but I could go either way.

Comment (bejado) [reply filament/src/driver/DataReshaper.h]: True- in that case it's probably fine as is.

**Pull request merged**